osu.Game/Screens/Menu/Intro.cs

```diff
@@ -56,30 +62,66 @@ public Intro()
             };
         }
 
+        private Bindable<bool> menuVoice;
+        private Bindable<bool> menuMusic;
+        private TrackManager trackManager;
+        private BeatmapInfo beatmap;
+        private WorkingBeatmap song;
+        private int choosableBeatmapsetAmmout;
+
         [BackgroundDependencyLoader]
-        private void load(AudioManager audio)
+        private void load(OsuGameBase game, AudioManager audio, OsuConfigManager config, BeatmapDatabase beatmaps)
         {
-            welcome = audio.Sample.Get(@"welcome");
-            seeya = audio.Sample.Get(@"seeya");
+            menuVoice = config.GetBindable<bool>(OsuConfig.MenuVoice);
+            menuMusic = config.GetBindable<bool>(OsuConfig.MenuMusic);
+            if (!menuMusic)
+            {
+                trackManager = game.Audio.Track;
+                choosableBeatmapsetAmmout = beatmaps.Query<BeatmapSetInfo>().Count();
+                if (choosableBeatmapsetAmmout > 0)
+                {
+                    beatmap = beatmaps.GetWithChildren<BeatmapSetInfo>(RNG.Next(1, choosableBeatmapsetAmmout)).Beatmaps[0];
+                    song = beatmaps.GetWorkingBeatmap(beatmap);
+                    Beatmap = song;
+                }
+            }
 
             bgm = audio.Track.Get(@"circles");
             bgm.Looping = true;
+
+            welcome = audio.Sample.Get(@"welcome");
+            seeya = audio.Sample.Get(@"seeya");
+
         }
 
         protected override void OnEntering(Screen last)
         {
             base.OnEntering(last);
 
-            welcome.Play();
+            if(menuVoice)
+                welcome.Play();
 
             Scheduler.AddDelayed(delegate
             {
-                bgm.Start();
+                if(menuMusic)
+                    bgm.Start();
+                else if (song != null)
+                {
+                    Task.Run(() =>

```

Comment:
we can't just be carelessly adding all this logic all over the place. when you see something used in more than one places, don't just copy and paste; refactor.